src/bytes_mut.rs

```diff
@@ -139,7 +103,25 @@ impl BytesMut {
     /// ```
     #[inline]
     pub fn with_capacity(capacity: usize) -> BytesMut {
-        BytesMut::from_vec(Vec::with_capacity(capacity))
+        unsafe {
+            if capacity == 0 {
+                BytesMut {
+                    data: ptr::null_mut(),
+                    ptr: NonNull::new_unchecked(&GUARD as *const u8 as *mut u8),
```

Comment:
Why not just `NonNull::dangling`?

Reply:
I just wasn't aware of it. Updated to use that now instead